### ISSUE TITLE
Windows mingw32 install error

### ISSUE THREAD

**Issue by Grillard** [ing-bank/sparse_dot_topn#19]
Hello, when trying to install sparse_dot_topn in windows with a mingw32 compiler I came across this error:
```
    C:\msys64\mingw64\bin\gcc.exe -mdll -DMS_WIN64 -Wall -I./sparse_dot_topn -Ic:\python37\lib\site-packages\numpy\core\include -Ic:\python37\include -Ic:\python37\include -c ./sparse_dot_topn/sparse_dot_topn.cpp -o build\temp.win-amd64-3.7\Release\.\sparse_dot_topn\sparse_dot_topn.o -Ox
    cc1plus.exe: error: argument to '-O' should be a non-negative integer, 'g', 's' or 'fast'
```

the mingw32 compiler was not happy with the -Ox argument passed at the end.

This can be easily solved by the using the default `extra_compile_args` in setup.py when os is nt but compiler is mingw32, which can be achieved by adding `os.compiler['CC']!='mingw32'` to the if condition:

```
if os.name == 'nt' and os.compiler['CC']!='mingw32':
    extra_compile_args = ["-Ox"]
else:
    extra_compile_args = ['-std=c++0x', '-pthread', '-O3']
```

I did this in a fork I did of this repo and I am successfully using the library!

Hope this helps! Cheers!

G

**Comment by ymwdalex** [comment]
@Grillard thanks for the solution for solving mingw32 complier failure.

Question about your solution:
Should `if os.name == 'nt' and os.compiler['CC']!='mingw32'` be `and` or `or`?

**Comment by Grillard** [comment]
it should be `and.` If you put `or` every compiler except mingw32 will use the Windows arguments. Also, I think mingw32 is available only in Windows!

**Comment by ymwdalex** [comment]
@Grillard I created this PR following your suggestion. Could you please review it? Thanks!

**Comment by Grillard** [comment]
Hello,

That solution looks fine to me!

Cheers,

G

**Comment by ymwdalex** [comment]
@Grillard merged, thanks!

**Comment by ymwdalex** [comment]
@Grillard this solutation didn't work because os does not have compiler members. Do you have any other suggestion?

**Comment by Sandy4321** [comment]
Will it work for Windows with anaconda?

**Comment by Sandy4321** [comment]
Since anaconda uses Microsoft visual studio as compiler

**Comment by ymwdalex** [comment]
@Sandy4321 @Grillard 

Thanks for involving the discussion of this thread! We just released [v0.3.3](https://pypi.org/project/sparse-dot-topn/#files). We employed Github actions and CiBuildWheel, and make the wheels available for Py3.6-3.10 on multiple architectures of Linux, MacOS and Windows.

If you still have installation problems, please try the latest version with pip install.

If you find any problem with v0.3.3, please let us know. Thanks!

**Comment by ymwdalex** [comment]
[v0.3.3](https://pypi.org/project/sparse-dot-topn/#files) should fix this issue
